stdlib/public/core/RangeReplaceableCollection.swift.gyb

```diff
@@ -1142,11 +1142,11 @@ extension RangeReplaceableCollection
 public func + <C : RangeReplaceableCollection, S : Sequence>(lhs: C, rhs: S) -> C
   where S.Iterator.Element == C.Iterator.Element {
 
-  var lhs = lhs
```

Comment:
Even if it doesn't happen now, we want it to eventually be able to happen (and @gottesmm's changes will make it more likely to happen). This new implementation will make it impossible to ever happen. If we want to accommodate reference-semantics collections, maybe `RangeReplaceableCollection` should have a `copy()` method that's a no-op returning `self` by default.